cpp/include/cudf/detail/scatter.hpp

```diff
@@ -135,6 +135,12 @@ std::unique_ptr<table> boolean_mask_scatter(
     rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource(),
     cudaStream_t stream = 0);
 
+std::unique_ptr<column> boolean_mask_scatter(
+    column_view const& input, column_view const& target,
+    column_view const& boolean_mask,
+    rmm::mr::device_memory_resource *mr,
+    cudaStream_t stream = 0);
```

Comment:
Is this overload really needed or can we just use `table_view` for source and target?

Reply:
But for `mask`/`where`, the comparison has to happen column-to-column basis as opposed to the entire table vs a column of a boolean mask, so I thought we should expose this method and did some cleanup of existing cython binding. 
But this cython API isn't actually being used in my code because of the reason the `i`th scalar in `input` will be written to all columns of the `output` table at the location of the `i`th true value in `boolean_mask`.

For example:
This is what `boolean_mask_scatter` currently does:
```c++
* input: {{1, 5, 6, 8, 9}}
 * boolean_mask: {true, false, false, false, true, true, false, true, true, false}
 * target:       {{   2,     2,     3,     4,    4,     7,    7,    7,    8,    10}}
 *
 * output:       {{   1,     2,     3,     4,    5,     6,    7,    8,    9,    10}}
```

But we actually want the functionality of `copy_if_else` where we pick values from lhs/rhs based on True/False.
 https://github.com/rapidsai/cudf/blob/branch-0.14/cpp/include/cudf/copying.hpp#L452

Reply:
Had a discussion offline and since we can construct `Table` from a `colum` using `as_frame()`, decided to remove the newly introduced overload.